maintenance/doMaintenance.php

```diff
@@ -100,6 +100,16 @@
 	require( MWInit::interpretedPath( 'AdminSettings.php' ) );
 }
 $maintenance->finalSetup();
+
+// check if specific task is already running (in case k8s will accidentally create two instances of cron job)
+if ($maintenance->status->isRunning()) {
```

Comment:
I wonder if we should do this for every execution of maintenance scripts (since it'll include also manually started ones). Maybe we should check an env var like `IS_MAINTENANCE_JOB` which would be set only inside cronjob containers, so that manual runs are not affected?

Reply:
I think we only need to make this work for just a few maintenance scripts. What if we just extend `Maintenance` class to handle locking mechanism and explicitly use it in specific cronjob scripts?

Reply:
the purpose of doing this in `Maintenance` was to make it as generic as possible, since k8s cronjobs can be created more than once no matter what job is running. also having any differences in `$maintenance` would require to add `if` check for every `$status` invocation. 

on the other hand, if we'll agree on extending `Maintenance` class, which scripts should I extend?